### PR TITLE
BinaryFormatter security updates

### PR DESCRIPTION
**Pull request by Rick-Anderson**
See https://github.com/dotnet/docs/issues/19460
@gewarren does this look like the right approach? How do I preview the build?

### CODE REVIEW

**Comment by PRMerger17** [comment]
@Rick-Anderson : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by PRMerger6** [comment]
@Rick-Anderson : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/includes/fix-binaryformatter.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Use a secure serializer instead, and **don't allow an attacker to specify an arbitrary type to deserialize**. For more information see the [Preferred alternatives](/dotnet/standard/serialization/binaryformatter-security-guide#preferred-alternatives).
+- Make the serialized data tamper-proof. After serialization, cryptographically sign the serialized data. Before deserialization, validate the cryptographic signature. Protect the cryptographic key from being disclosed and design for key rotations.
```
Comment (Rick-Anderson) [inline docs/code-quality/includes/fix-binaryformatter.md:2]: @GrabYourPitchforks I removed the following, should I add it back?

Restrict deserialized types. Implement a custom System.Runtime.Serialization.SerializationBinder. Before deserializing with BinaryFormatter, set the Binder property to an instance of your custom SerializationBinder. In the overridden BindToType method, if the type is unexpected, throw an exception to stop deserialization.
Ensure that all code paths have the Binder property set.

**Comment by dotpaul** [comment]
> See [dotnet/docs#19460](https://github.com/dotnet/docs/issues/19460)
> @gewarren does this look like the right approach? How do I preview the build?

Hey Rick, you want to submit your PR to the visualstudio-docs-pr repo.

**Comment by ktoliver** [comment]
Hi @Rick-Anderson - This pull request was opened in the public repo. Would you submit these updates in the private repo? We'll close this PR. Thanks.

<hr>

Related links in the [internal Microsoft Docs contributor guide](https://review.docs.microsoft.com/help/contribute/?branch=master):

[**Public vs. private repos**](https://review.docs.microsoft.com/help/contribute/contribute-authors-in-public-repo?branch=master) | [**Move a commit to a different repo (public -> private)**](https://review.docs.microsoft.com/help/contribute/contribute-how-to-manage-act-guidelines-public-pr?branch=master#pull-public-commits-to-the-private-repo-for-updates-staging-publication) | [**GitHub for content reviews**](https://review.docs.microsoft.com/help/contribute/contribute-how-to-write-pull-request-reviews?branch=master)
[**Quick reference for contributing to the docs**](https://review.docs.microsoft.com/help/contribute/contribute-quick-reference?branch=master) | [**Make changes to a Docs article**](https://review.docs.microsoft.com/help/contribute/edit-doc-with-vs-code-and-git-cli?branch=master) | [**PR review quality criteria**](https://review.docs.microsoft.com/help/contribute/contribute-how-to-write-pull-request-quality-criteria?branch=master)

**Comment by Rick-Anderson** [comment]
@ktoliver 

$ git clone https://github.com/MicrosoftDocs/visualstudio-docs-pr.git
Cloning into 'visualstudio-docs-pr'...
remote: The `MicrosoftDocs' organization has enabled or enforced SAML SSO. To access
remote: this repository, visit https://github.com/enterprises/microsoftopensource/sso?authorization_request=AA3QG5GWXN4A2PB3BLMVTMK7BY4E7A5PN5ZGOYLONF5GC5DJN5XF62LEZYAVOASZVVRXEZLEMVXHI2LBNRPWSZGODG2D5ZFPMNZGKZDFNZ2GSYLML52HS4DFVNHWC5LUNBAWGY3FONZQ
remote: and try your request again.
fatal: unable to access 'https://github.com/MicrosoftDocs/visualstudio-docs-pr.git/': The requested URL returned error: 403

**Pull request closed without merging**